### ISSUE TITLE
BGL (Bitgesell) Bounty/Improvement Program ($50000 budget)

### ISSUE THREAD

**Issue by wu-emma** [BitgesellOfficial/bitgesell#81]
Bitgesell (BGL) project is open for the contributions and participation. 

BGL is BTC-derived cryptocurrency with own blockchain, transaction fee burning 
mechanics and faster halving rate. A good source of the information would be the 
first 2 posts in the bitcointalk forum thread: https://bitcointalk.org/index.php?topic=5238559.0 

A $50000 budget formed by the BGL community to motivate developers make technical contributions. 

Code improvements that would be paid include: 
- fixes for unit and integration tests (this process is approximately 80% completed).
  A good chunk of tests fixed (see example here https://github.com/BitgesellOfficial/bitgesell/pull/46
  would be compensated around $1000); 
- cleaning the codebase (this must require thorough understanding of what is required 
  and what is not), just fixing some comment links or introducing changes to readme files 
  could be useful (but bounty payment would be less); 
- we are also looking for maintainers to review PRs, test the builds, test changes 
  manually and running test suites;
- Continuous Integration: there are CirrusCI configs written to build the 
  software for various platforms, there are minor issues present and fixes/updates to 
  those are also welcome; 
- fixes to the UI are welcome, example: BGL has only segwit addresses support 
  (no legacy), so the feature of signing a message is not supported (BTC does not 
  support this too), removing this from UI in a graceful way would be considered as 
  useful contribution; 
- infrastructural upgrades: if you want to bring up BGL testnet (which is not supported 
  at the moment as the fees in mainnet are very reasonable) this would be much welcome too; 

This list is just an example, if you want to contribute in another way that you think 
brings value to the project, feel free to ask prior (if the task is large) or make a 
partial change/prototype. Honest contributions are welcome, the cheat/abuse contributions 
like formal changes (formatting, a pair of comments added, etc.) would be rejected. 
There are other repositories of the project (where you can propose updates too), notably: 
- BGL-ERC20 Bridge (https://github.com/BitgesellOfficial/wbgl-bridge); 
- BGL Web wallet (https://github.com/BitgesellOfficial/bitgesell-wallet); 

We are strong believers in open-source and the approach that GitCoin is offering to the 
worldwide developer community: freedom, transparency and direct cooperation. 
Please respect everyone and don't spam messages that make no sense, any questions 
could be asked in: 
- the comment section in this github issue; 
- the project forum thread (https://bitcointalk.org/index.php?topic=5238559.0);
- project Discord channels (https://discord.gg/Ubp359vZEF); 

PRs would be paid in USDT upon approval, contributions in other areas (like infrastructure) 
would be compensated after implementation. People that are interested are welcome to become regular 
maintainers of the project (this is negotiable after some contribution is made);

Welcome to Bitgesell team and good luck!

3 maintainers would process new PRs: @madnadyka @janus @van-orton please keep incoming requests processed in good tempo (accept, reject, comment, etc.).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 1000.0 USDT (1000.00 USD @ $1.0/USDT) has been submitted by__:

 1. [@seeergo](https://gitcoin.co/seeergo)
 2. [@alekcangp](https://gitcoin.co/alekcangp)
 3. [@leetdev](https://gitcoin.co/leetdev)
 4. [@naftalimurgor](https://gitcoin.co/naftalimurgor)
 5. [@coinslist](https://gitcoin.co/coinslist)
 6. [@0xswakei](https://gitcoin.co/0xswakei)
 7. [@gitcoindev](https://gitcoin.co/gitcoindev)
 8. [@ckirwa](https://gitcoin.co/ckirwa)



@wu-emma please take a look at the submitted work:
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge/pull/19) by @ckirwa
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge/pull/17) by @0xswakei
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/88) by @gitcoindev
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge/pull/16) by @coinslist
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge) by @naftalimurgor
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge/pull/14) by @leetdev
* [PR](https://github.com/BitgesellOfficial/wbgl-bridge/pull/11) by @seeergo
* [PR](https://github.com/alekcangp/bgl/blob/main/README.md) by @alekcangp

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/29030)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/29030).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $1,003,918,423.97 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 264 years from now.
Please review their action plans below:


**1) [seeergo](https://gitcoin.co/seeergo) has started work.**

I will add typescript to the front-end part of the bridge project
**2) [leetdev](https://gitcoin.co/leetdev) has started work.**

Add Metamask support to WBGL bridge.
**3) [coinslist](https://gitcoin.co/coinslist) has started work.**

I will add TypeScript to the back-end of the application. This includes adding strict typing and static types.
**4) [0xswakei](https://gitcoin.co/0xswakei) has started work.**

I will add unit tests to the front-end of https://github.com/BitgesellOfficial/wbgl-bridge repo
**5) [gitcoindev](https://gitcoin.co/gitcoindev) has started work.**

Improve and fix Bitgesell functional tests and the functional tests framework.
**6) [alekcangp](https://gitcoin.co/alekcangp) has started work.**

A live visualisation of Bitgesell network activity: http://mempool.bglnode.online.

https://github.com/alekcangp/bglpool
**7) [kryptos-ke](https://gitcoin.co/kryptos-ke) has started work.**

Hello,

I will update MetaMask feature to support Goerli Testnet in Development since Protcol changes on Ethereum deprecates Ropsten, Rinkeby Testnet.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/29030).

**Comment by janus** [comment]
> Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done
> 
> **Work has been started**.
> 
> These users each claimed they can complete the work by 264 years, 4 months from now. Please review their action plans below:
> 
> **1) [seeergo](https://gitcoin.co/seeergo) has started work.**
> 
> I will add typescript to the front-end part of the bridge project
> 
> Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/29030).

@seeergo
What is your scope of work?

**Comment by janus** [comment]
Could you add tests to your work?

**Comment by SeeeRGo** [comment]
@janus to answer both your questions. Scope of my work is entire "app" folder, it's not that big)
I can add tests no problem, but to save us some back and forth can you specify, do you have anything particular in mind or any tests I would deem necessary would be enough? Since the app is not that big I can get pretty high coverage % :)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 264 years, 4 months from now.
Please review their action plans below:


**1) [gitcoincode](https://gitcoin.co/gitcoincode) has started work.**

I will add typescript to the front-end part of the bridge project

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/29031).

**Comment by ayushkumar63123** [comment]
Please check my pull requests

**Comment by janus** [comment]
> @janus to answer both your questions. Scope of my work is entire "app" folder, it's not that big) I can add tests no problem, but to save us some back and forth can you specify, do you have anything particular in mind or any tests I would deem necessary would be enough? Since the app is not that big I can get pretty high coverage % :)

@SeeeRGo
**I would want you to do the followings:**
1) Add unit test.
2) Make sure that your code works with server(service).
3) Integrate with MetaMask : Ethereum and BSC.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__The funding of 1000.0 USDT  (1000.00 USD @ $1.0/USDT) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,230,270.15 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by JeremyZeng77** [comment]
Submitted a WBGL bridge improvement under this program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/34
- Scope: fixes async nonce initialization for ETH/BSC bridge sends, fixes `ETH_NONCE` env parsing, and documents nonce override variables.
- Verification: `git diff --check`, `node --check service/src/modules/chores.js`, `node --check service/src/utils/config.js`.

Payment if approved: USDT TRC20 `TWupTAoFjxR2VmqUUvzhfmUwXVZR6Kvz1Y`.

**Comment by JeremyZeng77** [comment]
Submitted a Bitgesell Wallet metadata/docs cleanup under this program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/2
- Scope: updates package metadata from the old `epexa/bitgesell-wallet` path to `BitgesellOfficial/bitgesell-wallet`, and normalizes README wording/link text.
- Verification: `git diff --check`; parsed `package.json` with Node.

Payment if approved: USDT TRC20 `TWupTAoFjxR2VmqUUvzhfmUwXVZR6Kvz1Y`.

**Comment by Sikkra** [comment]
Submitted a WBGL bridge frontend test improvement under this program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/35
- Scope: adds React Testing Library smoke coverage for the app shell, conversion tabs, and wallet connection guidance so `app` has an actual passing test suite instead of failing with `No tests found`.
- Verification: `npm install`, `CI=true npm test -- --watchAll=false` (`1 passed`), `git diff --check`.

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted another WBGL bridge improvement under this program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/36
- Scope: fixes the BGL-to-WBGL submit UX so backend errors are shown in-page instead of only logged to the console, addressing the Continue-button no-feedback symptom from wbgl-bridge#28.
- Verification: `CI=true npm test -- --watchAll=false` (`1 passed`), `git diff --check`, and `node --openssl-legacy-provider ./node_modules/react-scripts/scripts/build.js` (compiled with existing warnings).

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted another WBGL bridge improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/37
- Scope: Adds a pre-submit WBGL-to-BGL bridge fee preview with a regression test.
- Related bridge issue: https://github.com/BitgesellOfficial/wbgl-bridge/issues/22
- Verification: frontend test passes, diff check passes, production build compiles with existing warnings.

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted another WBGL bridge improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/38
- Scope: Covers submitted WBGL transaction error paths and adds focused helper tests.
- Related bridge issue: https://github.com/BitgesellOfficial/wbgl-bridge/issues/21
- Verification: service syntax checks pass, focused Mocha test passes (`4 passing`), diff check passes.

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted a Bitgesell web wallet workflow improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/3
- Scope: Makes `npm test`, `npm run build`, `npm run local-server`, and `npm run dist-start` usable from Windows/PowerShell without POSIX-only quoting or environment assignment syntax.
- Verification: `npm test` passes, `npm run build` completes and writes `bitgesell-wallet-dist`, Node syntax checks pass for the new scripts, and a local-server smoke request returned `HTTP/1.1 200 OK`.

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted another WBGL bridge improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/39
- Scope: Enforces the bridge's available BGL reserve before a user sends WBGL, preventing an avoidable return path when the bridge cannot cover the requested amount.
- Related bridge issue: https://github.com/BitgesellOfficial/wbgl-bridge/issues/20
- Verification: frontend test passes, diff check passes, production build compiles with existing warnings.

USDT-TRC20 payout address if accepted: `TPwPFww7zxXFQ7zugo22gktQhckWVarRqi`

**Comment by Sikkra** [comment]
Submitted another Bitgesell wallet bounty candidate: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/4

Summary:
- Re-enables request cleanup callbacks after fetch/network failures so the Send form is not left disabled.
- Escapes untrusted RPC/explorer error text before rendering it in SweetAlert HTML.
- Keeps the existing CORS help link behavior while sanitizing the underlying error message.

Verification:
- .\\node_modules\\.bin\\eslint.cmd --rule "linebreak-style: 0" src\\api.js src\\utils.js src\\send\\send.js
- node -e syntax parse check for src/utils.js, src/api.js, and src/send/send.js
- git diff --check

Payout address if approved: USDT TRC20 TPwPFww7zxXFQ7zugo22gktQhckWVarRqi

**Comment by Luzijano** [comment]
Submitted a small `gobglbridge` reliability/test-coverage improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/gobglbridge/pull/4
- Scope: fixes the misspelled `returnsuccess` Redis status set key (`brdigeops:returnsuccess` → `bridgeops:returnsuccess`) so returned-success bridge operations stay in the same `bridgeops:` namespace as the other statuses.
- Adds regression tests for the Redis status namespace and the exact `returnsuccess` mapping.

Validation:
- RED before fix: `go test ./config` failed on the misspelled key.
- GREEN after fix: `go test ./config`
- Full suite: `go test ./...`

This also supports the testing goals in BitgesellOfficial/gobglbridge#3.

**Comment by Luzijano** [comment]
Submitted another contributor/workflow reliability improvement under the bounty/improvement program:

PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/1

Summary:
- fixes fresh `npm ci` failing because the root npm script named `dependencies` is treated as an npm lifecycle script
- renames the service-start script to explicit `services:up`
- updates README instructions

Validation:
- reproduced failure before the fix with `rm -rf node_modules && npm ci`
- verified after fix with `rm -rf node_modules && npm ci`
- verified lint with `npm run lint`

**Comment by Luzijano** [comment]
Submitted another small official Docker/onboarding cleanup under the bounty/improvement program:

PR: https://github.com/BitgesellOfficial/bgld-docker/pull/1

Summary:
- updates README links from the original contributor fork to the official `BitgesellOfficial/bgld-docker` repo
- fixes a Bitgesell spelling typo
- fixes the documented test runner path from `tests/run.sh` to the actual `test/run.sh`

Validation:
- verified updated GitHub/raw links resolve
- ran shell syntax validation for scripts
- verified `./test/run.sh bgld-autopilot-test:latest` works
- prior build validation: `docker build -t bgld-autopilot-test .`

**Comment by Luzijano** [comment]
Submitted a small Go toolkit build/test reliability improvement under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/go-bitgesell-toolkit/pull/1
- Scope: restores `go test ./...` for `go-bitgesell-toolkit` by fixing the invalid Go directive, duplicate config type declarations, undefined response helper types, and the checked-in example package.
- Validation: RED before fix: `go test ./...` failed on the invalid Go directive and compile errors; GREEN after fix: `go test ./...`; `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by Luzijano** [comment]
Submitted another small Bitgesell package/docs metadata cleanup under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-bip39/pull/1
- Scope: updates `bitgesell-bip39` package metadata to the official repository, makes the description Bitgesell-specific, fixes the README browserify package name, and normalizes the BitcoinJS attribution.
- Validation: `npm test -- --runInBand` (383 tests passing); `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by Luzijano** [comment]
Submitted another small Rust CI reliability improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/rust-bitgesellcore-rpc/pull/1
- Scope: fixes the workflow env name (`RUSTFMTCHK` -> `RUSTFMTCHECK`) so the existing stable Rust CI job actually triggers `contrib/test.sh`'s rustfmt check branch.
- Validation: RED/proof before fix showed workflow/script env mismatch; GREEN/proof after fix confirmed the spelling matches; `bash -n contrib/test.sh`; `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by Luzijano** [comment]
Submitted another small Bitgesell docs/onboarding cleanup under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-utxo-dump/pull/1
- Scope: fixes a duplicated `wget` in the setup docs and updates the Go install command from the old contributor fork path to the official `BitgesellOfficial/bitgesell-utxo-dump` repository.
- Validation: RED/proof before fix showed the duplicated command and stale fork install path; GREEN/proof after fix confirmed the README now uses the official path; `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by Luzijano** [comment]
Submitted another small WBGL bridge SDK safety/test improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/1
- Scope: removes private-key logging from README/test examples (`evmPrivateKey` / `process.env.evmPrivateKey`) and adds a regression test to keep these examples from reintroducing EVM private-key logs.
- Validation: RED before fix: `npx jest src/__tests__/SecretLogging.test.ts --runInBand` failed on the README secret log; GREEN after fix: same command passes; `npx eslint src/__tests__/SecretLogging.test.ts src/__tests__/WBGL.test.ts --ext .ts`; `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by Luzijano** [comment]
Submitted another small Bitgesell package/docs metadata cleanup under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/jsbgl/pull/3
- Scope: updates `jsbgl` package repository/bugs/homepage metadata to the official `BitgesellOfficial/jsbgl` repo, refreshes README status/package links from stale `jsbtc`/`jsbtc.js` references to `jsbgl`, and fixes the test link to the current official repo/default branch path.
- Validation: RED/proof before fix showed stale `bitaps-com/jsbgl`, `jsbtc`, and `jsbtc.js` references; GREEN/proof after fix confirmed those stale refs were removed; package JSON parse check; `git diff --check`.

No payout address posted unless/until accepted or requested.

**Comment by DYSfu** [comment]
Submitted a small SDK reliability fix under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/1
- Scope: wires `SDKConfig.logger` into the Address, Blockchain, Mempool, and Transaction wrappers, and removes a stray `describe.only` so the full Jest suite runs.
- Validation: `npx jest src/__tests__/logger.test.ts --runInBand`; `npm run build:main`; `npm test -- --runInBand`; `git diff --check`.

**Comment by tolga-tom-nook** [comment]
Bounty/improvement submissions with RED/GREEN proof:

1. https://github.com/BitgesellOfficial/bgld/pull/4
   - Scope: focused Go test/build reliability fix in official `bgld` repo.
   - RED: upstream `go test ./mempool` fails on `t.Fatal` with `%d` vet error.
   - GREEN: PR branch `go test ./mempool` passes.

2. https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/2
   - Scope: SDK correctness fix in official TypeScript blockchain SDK.
   - RED/proof: upstream `getAddressUTXO()` calls the unconfirmed-transactions endpoint and returns the wrong data shape.
   - GREEN: dedicated endpoint regression test passes; `npx tsc -p tsconfig.json` passes; full Jest suite passes (`6` suites / `23` tests).

If approved for payout, USDT/EVM-compatible address: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

Assisted by Hermes Agent.

**Comment by tolga-tom-nook** [comment]
Additional bounty/improvement submission with RED/GREEN proof:

- https://github.com/BitgesellOfficial/gobglbridge/pull/5
- Scope: WBGL bridge robustness/security fix.
- RED: malformed but valid hex signature `0x00` panics in `validateMsgSignature` because code reads `sigBytes[64]` before checking decoded length.
- GREEN: short signatures now return a normal validation error; focused regression test passes; full `go test ./...` passes.

If approved for payout, USDT/EVM-compatible address: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

Assisted by Hermes Agent.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell improvement PR for the bounty program:

PR: https://github.com/BitgesellOfficial/go-bgld/pull/2

Scope: fixes `go-bgld` `GetBlockheader` decoding for Core-shaped JSON responses. Bitgesell Core returns `bits` as a compact hex string like `"1d00ffff"`, and the previous/next block hash fields use lower-case JSON keys. The PR updates the struct decoding and adds an httptest regression for that response shape.

Validation:
- RED proof on upstream with the new regression: `go test ./...` fails before the fix because `BlockHeader.Bits` is `uint32` while Core returns a string.
- GREEN proof after fix: `go test ./...` passes.

Payout address if approved: USDT/EVM-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitting another PR for the Bitgesell improvement bounty:

PR: https://github.com/BitgesellOfficial/bglapiserver/pull/1
Repo: `BitgesellOfficial/bglapiserver`

Fix: transaction API routes were accidentally nested under the `blocks_data` feature flag in `api/routes.py`, so enabling `transaction` without `blocks_data` skipped the core transaction lookup/proof endpoints. This PR dedents those registrations so the `transaction` option controls them independently.

Validation:

```text
PYTHONDONTWRITEBYTECODE=1 python3 -m pytest tests/test_route_registration.py -q
1 passed in 0.02s

Python AST syntax OK
git diff --check: clean
```

Payout address for USDT/EVM-compatible rails: 0x4a76c7E64C08cF29B59eFC640b4ada97A270d428

**Comment by tolga-tom-nook** [comment]
Submitted a wbgl-bridge-sdk fix under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/2
- Scope: fixes BGL -> WBGL bridge submit endpoint construction. The SDK default/config example uses `https://bglswap.com/app/`; the previous code appended `/submit/bgl`, producing `https://bglswap.com/app//submit/bgl`. The PR normalizes trailing slashes so both trailing- and non-trailing-slash bridge endpoints resolve to canonical `/app/submit/bgl`.
- Validation: focused Jest regression added and passing (`2 passed`), targeted ESLint passing, `git diff --check` clean.
- Note: full `npm run build` is currently blocked by a repo dependency/toolchain mismatch (`@types/node` disposable symbols vs pinned TypeScript 4.9), unrelated to this change.

Payout address if approved: EVM/BSC USDC `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted a bitgesell-blockchain-sdk fix under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/3
- Scope: fixes `getMempoolState()` to call the aggregate `/mempool/state` endpoint instead of the paginated `/mempool/transactions` endpoint, and strengthens the regression test so it awaits the SDK call and asserts state fields (`inputs`, `outputs`, `transactions`).
- RED proof: strengthened focused Jest test failed before the code fix because the method returned transaction-list fields (`count`, `page`, `list`) instead of state fields.
- GREEN proof: focused mempool-state test passed, full `mempool.test.ts` passed, full Jest suite passed (`5` suites / `22` tests), `npm run build` passed, and `git diff --check` is clean.
- Note: targeted ESLint currently reports pre-existing repo config/rule issues unrelated to this change (`functional/no-class` in the existing class-based file and test parser-project inclusion).

Payout address if approved: EVM/BSC USDC `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Claim/proof for Bitgesell improvement bounty:

PR: https://github.com/BitgesellOfficial/jsbgl/pull/4

Scope: fixes `jsbgl` Bech32/SegWit address validation so mixed-case addresses are rejected before checksum normalization, while all-lowercase and all-uppercase Bech32 remain valid per BIP173. The previous implementation lowercased the HRP/payload for checksum validation and could accept mixed-case BGL Bech32 strings.

Validation evidence:
- RED: added regression made `npm test` fail in `Address functions: isAddressValid` with `AssertionError: expected true to equal false` for mixed-case `Bgl1QW508D6QEJXTDG4Y5R3ZARVARY0C5XW7K0FY5A3`.
- GREEN: `npm test` passes (`68 passing`).
- Package/dist proof: package main `require('./')` rejects the mixed-case address and accepts lowercase/uppercase forms.
- `git diff --check` clean.

Payout address if approved and USDT/EVM-compatible payout is used: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell improvement PR for this program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/3
- Scope: restores inherited `config.bitcoin*` aliases to the Bitgesell service configs so legacy tests/consumers reference the same Bitgesell objects; adds a no-daemon regression test and normalizes package-lock root metadata.
- Validation: `./node_modules/.bin/eslint src test --quiet`, `npm test -- --grep 'Configuration aliases'`, and `git diff --check` all passed.
- Payout address if approved: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted PR for this bounty/improvement lane:

- PR: https://github.com/BitgesellOfficial/bglutil/pull/4
- Scope: restores the missing `github.com/bitgesellofficial/bgld` module dependency and refreshed sums so a clean `bglutil` checkout can resolve Bitgesell-specific `bgld/...` imports instead of failing before tests start.
- Validation: `GOTOOLCHAIN=local go test ./base58 ./bech32 ./gcs ./gcs/builder ./hdkeychain` passed; `git diff --check HEAD~1..HEAD` clean.
- Payout address if approved: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by tolga-tom-nook** [comment]
Submitted PR for bounty consideration: https://github.com/BitgesellOfficial/bitgesell-arm/pull/1

Scope: fixes `docker-entrypoint.sh` so option and `.conf` invocations are rewritten to the repository's actual `bgl_oneshot` command instead of `bgl`, preventing container startup failure for standard option/config-file forms. Adds focused regression coverage for both forms.

Verification: `bash test/tests/entrypoint.sh` passed; `git diff --check origin/main..HEAD` clean.

Payout address if approved: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by 1507819106zxzx-crypto** [comment]
Submitted a focused Bitgesell web wallet hardening PR under the improvement program: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/5

Scope: escapes transaction IDs and wallet addresses before rendering them into table HTML attributes/text, and URL-encodes explorer/hash-route path segments generated from those table values. This keeps abnormal local-storage/API values from breaking attributes or injecting markup into the wallet tables.

Verification:
- node --check src/utils.js
- node --check src/transactions/transactions.js
- node --check src/my-addresses/my-addresses.js
- npx eslint --rule 'linebreak-style: 0' src/utils.js src/transactions/transactions.js src/my-addresses/my-addresses.js
- git diff --check

Payout address if accepted: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by tolga-tom-nook** [comment]
Submitted another non-overlapping Bitgesell ecosystem fix for the improvement/PR bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-utxo-dump/pull/2
- Scope: restores a clean Go module build for `bitgesell-utxo-dump` by adding `go.mod`/`go.sum`, updating stale module imports to `github.com/BitgesellOfficial/bitgesell-utxo-dump`, and fixing a formatting misuse caught by `go test`.
- RED upstream proof: `go test ./...` fails on `origin/main` with `pattern ./...: directory prefix . does not contain main module or its selected dependencies`.
- GREEN proof: `go test ./...` passes on the PR branch across all packages; `git diff --check origin/main...HEAD` is clean.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by tolga-tom-nook** [comment]
Submitted a focused wallet reliability PR for this program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/7
- Scope: `fetchQuery` now runs its `callbackAlways` cleanup when `fetch()` rejects, preventing the send form from staying disabled after network/CORS/RPC failures.
- Validation: `npm test` passed and `git diff --check` is clean.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by newmattock** [comment]
Withdrawn: after posting, I saw that https://github.com/BitgesellOfficial/bitgesell-utxo-dump/pull/2 was submitted earlier with the same core module-build/import-path scope. I closed https://github.com/BitgesellOfficial/bitgesell-utxo-dump/pull/3 to avoid a duplicate bounty claim.

**Comment by tolga-tom-nook** [comment]
Submitted a focused bgld-docker entrypoint fix for Bitgesell improvement bounty consideration:

- PR: https://github.com/BitgesellOfficial/bgld-docker/pull/2
- Scope: fixes the stale `btc_oneshot` privilege-drop guard in `docker-entrypoint.sh` so standard option/config startup rewritten to `bgl_oneshot` is re-execed through `gosu BGL` when the container starts as root, rather than running directly as root.
- Validation: RED against upstream failed the new regression with `entrypoint executed bgl_oneshot directly instead of dropping privileges`; GREEN after fix: `sh -n docker-entrypoint.sh`, `sh -n test/entrypoint.sh`, `sh test/entrypoint.sh`, and `git diff --check` all pass.
- Payout address if accepted: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell ecosystem fix for bounty consideration:

- PR: https://github.com/BitgesellOfficial/php-bitgesellcore-rpc/pull/1
- Scope: restores the PHP RPC client's Composer/package identity and autoloadability by aligning `composer.json`, `Config`, tests, and provider class strings with the repo's `naftalimurgor\Bitgesell` namespace; also fixes the `BitgeselldResponse` typo/class-string mismatch and replaces the stale duplicated Go README with PHP usage docs.
- Validation: `python3 -m json.tool composer.json`, static stale-reference scan for old Denpa/Bitcoin and typo symbols, and `git diff --check` all passed. PHP/Composer were unavailable in this runner, so PHPUnit could not be executed here.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/pybgl#3 to fix editable installs for `pybgl`.

Acceptance criteria satisfied:
- Improves a Bitgesell repository with a focused packaging/build fix.
- Restores a clean install path that previously failed before tests could run.
- Keeps the change minimal and reviewable.

Evidence:
- PR: https://github.com/BitgesellOfficial/pybgl/pull/3
- Before fix: `pip install -e .` failed because `build_ext` tried to resolve `libsecp256k1` with `pkg-config` before the bundled C library was built.
- After fix: `pip install -e .` succeeds in a clean temp venv.
- After fix: `python tests.py` passes 6 tests.
- `git diff --check` passes.

Payment or review notes: please let me know if you want any adjustment before bounty review.

**Comment by tolga-tom-nook** [comment]
Submitted a focused pybgl repository-link cleanup PR for this bounty program:

- PR: https://github.com/BitgesellOfficial/pybgl/pull/5
- Scope: updates README/install docs/Sphinx metadata/package URL from stale `bitaps-com/pybgl` links to the canonical `BitgesellOfficial/pybgl` repo, including the issue tracker.
- Validation: `python3 -m compileall -q pybgl`; stale-link scan for `bitaps-com/pybgl` in README/setup/docs returned no matches; `git diff --check` passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/pybgl#4 to enforce Bech32/SegWit address case rules in `pybgl`.

Acceptance criteria satisfied:
- Improves an official Bitgesell repository with a focused validation correctness fix.
- Rejects mixed-case Bech32 addresses before checksum normalization while preserving valid lowercase and uppercase Bech32 forms.
- Adds focused regression coverage and wires it into the existing `tests.py` runner.
- Keeps the change separate from BitgesellOfficial/pybgl#3, which only addressed editable-install build ordering.

Evidence:
- PR: https://github.com/BitgesellOfficial/pybgl/pull/4
- Validation: `python3 -m unittest pybgl.test.bech32_address_validation`
- Validation: `python3 tests.py`
- Validation: `git diff --check`

Payment or review notes: please let me know if you want any adjustment before bounty review.

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/electrumx#1 to add Bitgesell mainnet and testnet coin parameters to ElectrumX.

Acceptance criteria satisfied:
- Improves an official Bitgesell repository with a focused integration/configuration patch.
- Adds `Bitgesell` and `BitgesellTestnet` coin classes using parameters from Bitgesell Core chainparams.
- Adds focused regression coverage so the new coin classes are discoverable through `Coin.lookup_coin_class()` and match the Core address/RPC/genesis parameters.
- Keeps the change scoped to coin definitions and tests.

Evidence:
- PR: https://github.com/BitgesellOfficial/electrumx/pull/1
- Validation: `PYTHONPYCACHEPREFIX=/private/tmp/bitgesell-pycache python3 -m py_compile electrumx/lib/coins.py tests/lib/test_bitgesell_coin.py`
- Validation: `PYTHONDONTWRITEBYTECODE=1 /private/tmp/electrumx-bitgesell-venv/bin/python -m pytest tests/lib/test_bitgesell_coin.py -q` (`2 passed`)
- Validation: `PYTHONDONTWRITEBYTECODE=1 /private/tmp/electrumx-bitgesell-venv/bin/python -m pytest tests/lib/test_coins.py -q` (`16 passed`)
- Validation: `git diff --check`

Payment or review notes: please let me know if you want any adjustment before bounty review.

**Comment by tolga-tom-nook** [comment]
Submitted another focused wallet safety/reliability PR for Bitgesell improvement bounty consideration:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/8
- Scope: hardens the send form preflight so submit-time validation rechecks the recipient address, recalculates spend/new-balance state, and refuses to call `sendrawtransaction` when no serialized transaction has been generated. This prevents null/stale transaction RPC submissions if the form is submitted before the earlier field-change validation path runs.
- Validation: `node --check src/send/send.js`, `npm test`, and `git diff --check` all passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by tolga-tom-nook** [comment]
Submitted another focused bglapiserver README cleanup for Bitgesell bounty consideration:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/3
- Scope: refreshes stale README links from `bitaps-com/pybgl` and `bitaps-com/bglapiserver` to the canonical Bitgesell repositories, replaces Bitcoin-specific API module wording with Bitgesell-specific wording, and fixes nearby README grammar.
- Validation: `python3` stale-link/wording scan passed; `git diff --check` passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

Assisted by Hermes Agent.

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/multichain#1 to make the local multichain test path handle required Git submodules.

Acceptance criteria satisfied:
- Improves an official Bitgesell repository with a focused contributor/test-runner reliability patch.
- Addresses the clean-clone failure mode where `chain/filecoin/filecoin-ffi` and `chain/solana/solana-ffi` are uninitialized, causing Go module replacement paths such as `github.com/renproject/solana-ffi` to resolve to missing local files before tests can run.
- Keeps the change scoped to test setup and README instructions.

Evidence:
- PR: https://github.com/BitgesellOfficial/multichain/pull/1
- Validation: `bash -n test.sh`
- Validation: `git diff --check`
- Reproduction evidence: a plain clone showed uninitialized FFI submodules, and the Go module loader failed on missing `chain/solana/solana-ffi/go.mod`.

Payment or review notes: please let me know if you want any adjustment before bounty review.

**Comment by CharlieLZ** [comment]
Submitted a focused JS RPC client reliability fix under the Bitgesell improvement program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/4
- Scope: response logging no longer throws when a JSON-RPC batch response contains an unmatched id; matched sensitive results are still obfuscated and unmatched response entries are preserved unchanged.
- Regression coverage: added batch-response tests for matched, unmatched, and mixed response ids.
- Verification: focused Mocha regression passed, `npm run lint` passed, and `git diff --check HEAD~1..HEAD` passed.

Payment details can be provided if accepted or requested.

**Comment by CharlieLZ** [comment]
Submitted another focused Bitgesell docs reliability improvement under the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk-docs/pull/1
- Scope: fixes generated SDK documentation and package metadata links that still pointed at the unavailable personal fork `naftalimurgor/bitgesell-blockchain-sdk`, and replaces a stale GitHub Pages docs URL with the official docs repository URL.
- Validation: confirmed the old source repository returns 404, representative official source links return HTTP 200, the official docs repository exists, stale-link scan returns no matches, and `git diff --check` is clean.

Payment details can be provided if accepted or requested.

**Comment by CharlieLZ** [comment]
Submitted another improvement PR for the bounty:

- https://github.com/BitgesellOfficial/keep-core-bgl/pull/1

This adds a local Go workspace so normal checkouts can run focused Go tests using the vendored `btcd` and `keep-common` trees. Verified `go test ./cmd ./config` passes after the change.

Payment details can be provided if accepted or requested.

**Comment by CharlieLZ** [comment]
Submitted another focused Bitgesell improvement PR for bounty consideration:

- https://github.com/BitgesellOfficial/keep-core-bgl/pull/2

Scope: removes hard-coded `fmt.Println("Keccak256")` calls from the vendored `btcd/txscript` signature hash paths. These functions are library code used by signing/validation, so writing to stdout pollutes CLI output, tests, and downstream consumers.

Validation:
- RED before fix: focused regression failed with five `Keccak256` stdout lines.
- GREEN after fix: `go test ./txscript -run 'TestSignatureHashFunctionsDoNotWriteStdout|TestHashCache' -count=1` passes.
- `git diff --check` passes.

Payment details can be provided if accepted or requested.

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/go-bgld#3 to make RPC timeout handling cancel the underlying HTTP request instead of only returning from a timer goroutine.

Acceptance criteria satisfied:
- Improves an official Bitgesell repository with a focused reliability patch.
- Keeps the existing public timeout error string for callers.
- Adds/updates regression coverage so the timeout path remains verified.
- Reduces the timeout regression from a 30-second wait to a focused 1-second case, making the suite faster and less flaky.

Evidence:
- PR: https://github.com/BitgesellOfficial/go-bgld/pull/3
- Validation: `go test ./...` passed.
- Validation: `git diff --check` passed.

Payment or review notes: please let me know if you want any changes or need payout details after acceptance.

**Comment by newmattock** [comment]
Bounty: Bitgesell improvement program, BitgesellOfficial/bitgesell#81

Work completed: opened BitgesellOfficial/go-bitgesell-toolkit#3 to restore the official Go toolkit build/test path and make the address API methods decode typed responses.

Acceptance criteria satisfied:
- Improves an official Bitgesell repository with a focused build and test reliability patch.
- Fixes current `go test ./...` compilation blockers: duplicate `SDKConfig`, undefined `object` types, invalid example imports/usages, and missing transaction client fields.
- Adds the `setup` and `compile` Make targets already referenced by the repository GitHub Actions workflow.
- Expands address tests from no-error checks to decoded-value assertions.

Evidence:
- PR: https://github.com/BitgesellOfficial/go-bitgesell-toolkit/pull/3
- Baseline before fix: `go test ./...` failed at compile time.
- Validation after fix: `go test ./...` passed.
- Validation after fix: `make setup && make compile` passed.
- Validation after fix: `git diff --check` passed.

Payment or review notes: please let me know if you want changes or need payout details after acceptance.

**Comment by tolga-tom-nook** [comment]
Submitted another focused JS RPC client fix for Bitgesell bounty consideration:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/7
- Scope: fixes `Client#command()` multiwallet routing for mixed-case RPC method names. Direct calls like `command('getBalance')` were sent as lowercase `getbalance`, but the wallet-support lookup used the original mixed-case key and missed multiwallet support, so requests posted to `/` instead of `/wallet/<wallet>`.
- Regression coverage: single and batch mixed-case wallet commands now assert the selected wallet URI is used.
- RED proof on `origin/main`: new tests failed with `AssertionError: expected '/' to be '/wallet/wallet.dat'`.
- GREEN proof after fix: `./node_modules/.bin/mocha test/command_wallet_routing_test.js --timeout 20000 --require should` passed (`2 passing`); `./node_modules/.bin/eslint src/index.js test/command_wallet_routing_test.js` passed; `git diff --check` clean.
- Payout address if accepted: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

Assisted by Hermes Agent.

**Comment by jtc268** [comment]
Submitted a focused Bitgesell JS RPC client metadata/docs reliability fix for bounty consideration:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/8
- Scope: points package metadata and README setup docs at the official repo, removes retired fork/Docker image URLs that return 404, and aligns BGLd/BGL.conf/default port docs with current client defaults.
- Evidence: naftalimurgor/bitgesell-core returned 404; Docker Hub ruimarinho/bitgesell-core returned 404; src/index.js maps mainnet to 8454 and testnet to 18332.
- Validation: npm ci --ignore-scripts; focused mocha metadata test 2 passing; npm test -- --grep 'Package metadata' 2 passing; npm run lint -- --quiet passed; stale-link scan over README/package files clean.

Payment details can be provided if accepted or requested.

**Comment by tolga-tom-nook** [comment]
Hi team — quick maintainer follow-up on this bounty lane.

I submitted PR #5 in `BitgesellOfficial/gobglbridge` for issue #81 with RED/GREEN proof and minimal scope (`validateMsgSig` guard fix for malformed WBGL bridge signatures). Could you confirm whether this PR is still under review / still eligible, and if there are any remaining maintainer requirements before acceptance?

We’ve been actively pushing targeted fixes and can keep the batch low-noise/low-latency if this lane is still open.

**Comment by MyTH-zyxeon** [comment]
Hi Bitgesell maintainers — I’m interested in taking one of the paid items from this program (starting with a small, well-scoped PR such as fixing a specific failing unit/integration test).

Before I begin, can you confirm the current bounty process?

1) Where is the authoritative list of paid tasks (issues/labels/board)?
2) What is the current payment mechanism (USD / crypto / escrow) and who approves payouts?
3) Should contributors claim an issue first (and if so, what format), or submit a PR directly?
4) What acceptance criteria do you require for “test fixes” bounties (CI target, reproduction steps, coverage expectations)?
5) Are there any constraints around refactors vs. minimal fixes, or preferred target areas?

If you can point me to the current “how to claim + how to get paid” doc or a single issue you’d like me to start with, I can propose a concrete PR plan and timeline.

**Comment by MyTH-zyxeon** [comment]
Hi Bitgesell maintainers - I can take a first small paid slice from this program if it matches your current priorities.

Proposed initial PR scope:

- Pick one deterministic unit/integration test or CI failure in `BitgesellOfficial/bitgesell`.
- Keep the patch minimal and avoid consensus, wallet-format, or broad refactor changes unless the failing test requires it.
- Include RED/GREEN evidence in the PR: failing command or CI target before the fix, passing command after the fix, and a short explanation of why the change is safe.
- If the current best paid lane is not a test failure, I can instead take the UI clean-up item mentioned in this issue: gracefully remove or disable unsupported message-signing UI for BGL segwit-only behavior, with a focused regression check.

Before I start, could you confirm which of these two starting slices you prefer for bounty review?

1. test/CI fix in `BitgesellOfficial/bitgesell`
2. message-signing UI clean-up

No payment details are needed from my side until you confirm the preferred slice and acceptance path.

**Comment by MyTH-zyxeon** [comment]
Maintainers — quick proposal for the Bitgesell improvement program (#81).

I can take on a *small, reviewer-friendly* slice next (aiming for 1–2 narrow PRs):

1) **Unit/integration test fixes**: pick a handful of currently-failing tests (or flaky ones) and fix them with minimal code changes, with CI green.
2) **CI/CirrusCI polish**: address any minor CirrusCI config issues that prevent reliable multi-platform builds.

I already left a review-assist note on PR #276 to help get that one merged cleanly; happy to follow up with additional fixes once you point me at the highest-priority failing tests/CI pain points.

If you have a preferred starting point, please reply with either:
- a link to the CI run / failing test list you want tackled first, or
- a short priority list (tests vs CI vs maintainership chores).

I’ll keep changes narrow (no formatting-only noise) and split work into separate PRs if that helps review speed.

**Comment by tolga-tom-nook** [comment]
Submitted a new Bitgesell API contribution for bounty review:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/8
- Fix: impossible block-header `count` validation (`count < 1 and count > 2000`) changed to a working lower/upper bound guard, with regression coverage for `0` and `5000` count values.
- Requested liquid payout rail if accepted: USDT / USDC-compatible EVM address `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

Thank you — happy to adjust if maintainers prefer a different count handling behavior.

**Comment by MyTH-zyxeon** [comment]
Maintainers - quick queue summary for the Bitgesell bounty/improvement lane (#81).

As of 2026-06-02 UTC, there are at least three small, reviewer-friendly PRs open across the Bitgesell repos:

- `BitgesellOfficial/bitgesell#276` - `mergeStateStatus=CLEAN`
  - cuckoocache bool-fuzz compile fix
  - I already left a review-assist note here with suggested acceptance checks: https://github.com/BitgesellOfficial/bitgesell/pull/276#issuecomment-4591354994

- `BitgesellOfficial/gobglbridge#5` - `mergeStateStatus=CLEAN`
  - short WBGL signature panic/hardening fix
  - author has since added additional malformed-signature + recovery-ID coverage in follow-up comments

- `BitgesellOfficial/bglapiserver#8` - `mergeStateStatus=CLEAN`
  - block-header `count` validation fix with regression coverage
  - newly submitted here for bounty review: https://github.com/BitgesellOfficial/bitgesell/issues/81#issuecomment-4597188426

If you want to keep this lane moving without duplicate work, could you confirm one of these two routing choices?

1. Which repo/PR should be treated as the highest-priority review lane right now (`bitgesell#276`, `gobglbridge#5`, or `bglapiserver#8`)?
2. Which repo would you prefer for the next *new* small contribution after that review queue clears?
   - `bitgesell` test/CI fix
   - `gobglbridge` bridge hardening follow-up
   - `bglapiserver` API/test fix

Once you point at the preferred lane, I can keep future changes narrow and avoid duplicating work that is already waiting in review.

**Comment by MiguelKorTes** [comment]
Submitted a focused SDK fix for the bounty/improvement program: BitgesellOfficial/bitgesell-blockchain-sdk#6

Summary: getAddressUTXO() now calls /address/utxo/{address} instead of the unconfirmed transactions endpoint, describe.only was removed from address tests, and a mocked URL regression test was added.

Verification: npm run build; npm test -- --runTestsByPath src/__tests__/address-url.test.ts --coverage=false

Payout details can be provided after review/approval.

**Comment by MyTH-zyxeon** [comment]
Maintainers - quick routing refresh for the Bitgesell bounty/improvement lane (#81).

As of 2026-06-02 UTC, I can publicly verify at least four small, reviewable PRs waiting across the Bitgesell repos:

- `BitgesellOfficial/bitgesell#276`
  - cuckoocache bool-fuzz compile fix
  - `mergeStateStatus=CLEAN`
  - prior review-assist note: https://github.com/BitgesellOfficial/bitgesell/pull/276#issuecomment-4591354994

- `BitgesellOfficial/gobglbridge#5`
  - WBGL signature validation hardening
  - `mergeStateStatus=CLEAN`

- `BitgesellOfficial/bglapiserver#8`
  - block-header `count` validation fix with regression coverage
  - submitted here earlier today: https://github.com/BitgesellOfficial/bitgesell/issues/81#issuecomment-4597188426

- `BitgesellOfficial/bitgesell-blockchain-sdk#6`
  - `getAddressUTXO()` endpoint correction + regression test
  - newly submitted here: https://github.com/BitgesellOfficial/bitgesell/issues/81#issuecomment-4601674695

To keep this program moving without duplicate contributor effort, could you confirm:

1. Which of these PRs should be treated as the highest-priority review lane right now?
2. Which repo should contributors target next for fresh small bounty work once that lane clears?
   - `bitgesell`
   - `gobglbridge`
   - `bglapiserver`
   - `bitgesell-blockchain-sdk`

If you point at the preferred lane/repo, I can keep future work narrow and avoid piling more unreviewed parallel fixes onto the wrong surface.

**Comment by tolga-tom-nook** [comment]
Quick update from Tom Nook lane: I have 9+ clean, open PRs covering issue #81 with wallet-compatible fixes and regression coverage. I can keep this lane constrained to one narrow follow-up PR until we get maintainer priority. Please confirm priority repo/PR lane (bitgesell / gobglbridge / bglapiserver / bitgesell-blockchain-sdk / etc.) for review/payout sequencing.

**Comment by ColumbusLabs** [comment]
Submitted a focused SDK fix for the bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/7
- Scope: fixes `getBlockHeaders()` so omitting the optional `count` argument calls `/block/headers/{height}` instead of `/block/headers/{height}/undefined`, while preserving the counted `/block/headers/{height}/{count}` form.
- Regression coverage: added a mocked URL construction test for both omitted-count and provided-count calls.

Validation:
- `npm ci --legacy-peer-deps`
- `npm run build`
- `npm test -- --runTestsByPath src/__tests__/blockchain-url.test.ts --coverage=false`
- `git diff --check`

Payment details can be provided after review/approval.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell API contribution for bounty review:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/9
- Fix: `GET /blocks/last/{n}` and `GET /blocks/data/last/{n}` now reject zero, negative, and non-numeric counts with `PARAMETER_ERROR` instead of falling through to generic internal-error behavior or the model layer.
- Validation: `python3 -m py_compile api/views/blocks.py`; synthetic handler check verified `0`, `-1`, and `abc` return HTTP 400 for both affected handlers without calling model functions.

**Comment by tolga-tom-nook** [comment]
Submitted another small Bitgesell improvement PR for the bounty program:

PR: https://github.com/BitgesellOfficial/electrumx/pull/2

Fix: `electrumx` mempool tests passed `datetime.date.today().toordinal` (the bound method) to `random.seed`, which raises a `TypeError` on Python 3.12 during test collection. The PR calls `toordinal()` so the seed is the intended integer value.

Validation: `.venv/bin/python -m pytest -q tests/server/test_mempool.py` → `11 passed, 3 warnings in 1.02s`.

Payout rail if accepted: USDT/BSC `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by Cycle1337** [comment]
Submitted a focused Bitgesell packaging fix under the improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell/pull/358
- Related issue: #357
- Scope: removes obsolete/development packages from Debian runtime dependencies and keeps required development libraries in `Build-Depends`.
- Why it matters: the current package metadata pins `perl-modules-5.34`, which blocks installation on newer Ubuntu releases, and pulls development libraries for end users.
- Validation: static Debian control checks for runtime/build dependencies plus `git diff --check`.

Payment details can be provided after maintainer review/approval.

**Comment by nexicturbo** [comment]
Submitted two focused Bitgesell improvement PRs for bounty-program review:

- PR #359: https://github.com/BitgesellOfficial/bitgesell/pull/359
  - Scope: makes `test/lint/lint-includes.py` pass by removing duplicate `src/span.h` includes, removing obsolete `boost/process.hpp` from the expected Boost include list, and converting lint-reported include syntax.
  - Validation: `python test\lint\lint-includes.py`; `git diff --check`.

- PR #360: https://github.com/BitgesellOfficial/bitgesell/pull/360
  - Scope: makes `test/lint/lint-format-strings.py` runnable and green by invoking the helper through the active Python interpreter, updating the Bitgesell `BGL-tidy` exclusion, and excluding the generated secp precompute false-positive source.
  - Validation: `python test\lint\lint-format-strings.py`; `python -m py_compile test\lint\lint-format-strings.py test\lint\run-lint-format-strings.py`; `git diff --check`.

Both PRs are open, non-draft, and mergeable. Payment details can be provided after maintainer approval.
